cmd/wharf/run.go

```diff
@@ -204,6 +220,23 @@ func startWorkerServerWithCancel(ctx context.Context, store resultstore.Store) (
 	return ctx, server
 }
 
+func handleCancelSignals(f func()) {
+	waitForCancelSignal()
+	log.Info().WithDuration("gracePeriod", cancelGracePeriod).Message("Cancelling build. Press ^C again to force quit.")
+	go func() {
+		waitForCancelSignal()
+		log.Warn().Message("Received second interrupt. Force quitting now.")
+		os.Exit(2)
+	}()
+	f()
+}
+
+func waitForCancelSignal() {
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, os.Interrupt, os.Kill, syscall.SIGTERM, syscall.SIGHUP)
+	_ = <-ch
```

Comment:
No need for a write-only variable here. Just receiving the message is clear enough that we ignore the signal message.

```suggestion
	<-ch
```